plugins/integration-openai/src/static/providers/openai/openai.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from "@nestjs/common";
+import {
+  Configuration,
+  OpenAIApi,
+  ChatCompletionRequestMessage,
+  CreateChatCompletionRequest,
+} from "openai";
+
+export type CreateChatCompletionRequestSettings = Omit<
+  CreateChatCompletionRequest,
+  "model" | "messages"
+>;
+
+export {
+  ChatCompletionRequestMessage,
+  ChatCompletionRequestMessageRoleEnum,
+} from "openai";
+
+const CREATE_CHAT_COMPLETION_DEFAULT_SETTINGS: CreateChatCompletionRequestSettings =
+  {
+    temperature: 1,
+    max_tokens: 2048,
+    top_p: 1,
+    frequency_penalty: 0,
+    presence_penalty: 0,
+  };
+
+@Injectable()
+export class OpenaiService {
+  constructor() {}
+
+  async createChatCompletion(
+    model: string,
+    messages: ChatCompletionRequestMessage[],
+    requestSettings?: CreateChatCompletionRequestSettings
+  ): Promise<string> {
+    const configuration = new Configuration({
+      apiKey: process.env.OPENAI_API_KEY,
+    });
+    const openai = new OpenAIApi(configuration);
+
+    const settings = {
+      ...CREATE_CHAT_COMPLETION_DEFAULT_SETTINGS,
+      ...requestSettings,
+    };
+
```

Comment:
It would be better to move this initialisation at the constructor level to leverage the nestjs DI other then instantiating this `OpenAIApi` on every consumption

Reply:
These settings can differ per request 
For example I can use a different temperature per conversation etc

Reply:
Sorry I didn't mean to include the settings too. Just the instantiation of the openapi client